tests/unit/Foundation/PathsTest.php

```diff
@@ -48,7 +48,7 @@ public function it_derives_the_vendor_dir_from_the_base_path()
             'storage' => '/var/www/flarum/storage',
         ]);
 
-        $this->assertEquals('/var/www/flarum/vendor', $paths->vendor);
+        $this->assertEquals('/var/www/flarum'.DIRECTORY_SEPARATOR.'vendor', $paths->vendor);
```

Comment:
Why is only one of the slashes replaced? Shouldn't all of them be replaced?

Reply:
![image](https://user-images.githubusercontent.com/6401250/83063854-c575e880-a02e-11ea-8fbd-db96c394204c.png)

🤷

Reply:
Because of this: https://github.com/flarum/core/blob/3c87f800ddaa048d2c19ca296c8f1da9ffb3fbfa/src/Foundation/Paths.php#L37

I mean, we could also change the source code to just use a slash? That should still work on Windows, right?

Reply:
I would guess that it uses the defined paths above and appends to it, hence why only the last slash is the OS's slash type.

Reply:
Yes, but [it should not be necessary](https://stackoverflow.com/a/38428899/192741). So could we just change `Paths.php` to use a slash instead? Or does anyone see a downside to that?

Reply:
Oh I hadn't seen your reply until now, yeah windows works fine with either type of slashes so I don't think there'll be any problem with changing it.

Reply:
I'm fine with either, but if we can have consistent slashes everywhere, that would be preferable as long as it doesn't break anything. Which I'm asuming it doesn't since it currently works with both types of slashes, right?

Reply:
only case scenarios where it would be a problem is if you were using php functions/constants to get a path, and then for example for w/e reason exploding that path with `DIRECTORY_SEPARATOR` that wouldn't be a place to change to a forward slash since php would in windows return a path with backslashes.

Reply:
But wouldnt that issue still apply with the slashes that are already in the path?

Reply:
how so ?

Reply:
Well, we already have UNIX-style slashes in the `/var/www/flarum` component, so if you're trying to split on windows with `DIRECTORY_SEPARATOR`, it would fail anyway, right?

Reply:
yeah it would because the path is predefined, it would only properly work with the constant if the path was returned from a php function or a php constant such `__DIR__` because it would use the OS slash type.

Reply:
> Well, we already have UNIX-style slashes in the `/var/www/flarum` component.

That's just example data in our test case, though. [These](https://github.com/flarum/flarum/blob/3aeffd15aac9ef98dd44dc53eff960245ab0d247/site.php#L46-L50) would be realistic values, currently.

Reply:
Then we might as well go all out and make sure there's no literal slashes in the code. Or is that too extreme?

Reply:
I don't think it's necessary - and it would just make the code more verbose.

But I may be missing a case where the difference is relevant.

Reply:
So how do you want to proceed?

Reply:
[This was my proposal.](https://github.com/flarum/core/pull/2187#discussion_r431410375)